library/src/main/kotlin/com/vinted/actioncable/client/kotlin/Connection.kt

```diff
@@ -139,12 +123,9 @@ class Connection constructor(
 
         val request = requestBuilder.build()
 
-        val httpClient = httpClientBuilder.build()
-        val webSocketFactory = options.okHttpWebSocketFactory ?: httpClient
+        val webSocketFactory = options.webSocketFactory ?: OkHttpClient()
```

Comment:
... and apply it on fallback implementation:

```
?: OkHttpClient()
.newBuilder()
.sslSocketFactory(options.sslContext.socketFactory)
.hostnameVerifier(options.hostnameVerifier)
.build()
```
And inside documentation we could add explanation:

> Note: if you provide custom `WebSocket.Factory` implementation, then `sslContext` and `hostnameVerifier` options are ignored. You must set those options directly in your custom implementation instead of setting it on `options.connection`.